docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md

```diff
@@ -80,13 +83,346 @@ This example demonstrates how to configure the order state machine transition fr
+11. Stripe app confirms the PreOrder payment using the plugin: `\Spryker\Zed\Payment\Communication\Plugin\Checkout\PaymentConfirmPreOrderPaymentCheckoutPostSavePlugin`.
+  The `order_reference` is passed to the Stripe app to be connected with `transaction_id`.
+12. Stripe app processes the payment and sends a `PaymentUpdated` message to Spryker.
+  The message contains additional data, which you can see in the Back Office.
+13. Depending on payment status, one of the following messages is returned through an asynchronous API:
+  *  Payment is successful: `PaymentAuthorized` message.
+  *  Payment is failed: `PaymentAuthorizationFailed` message.
+14. Further on, the order is processed through the OMS.
+
+All payment related messages mentioned above are handled by `\Spryker\Zed\Payment\Communication\Plugin\MessageBroker\PaymentOperationsMessageHandlerPlugin`, which is registered in `MessageBrokerDependencyProvider`.
+
+### Example integration
+
+To check out how the integration works, see this [example application](https://github.com/spryker-projects/spa-checkout-glue-with-stripe).
+
+Before the customer is redirected to the summary page, all required data is collected: customer data, addresses, and selected shipment method. When the customer goes to the summary page, to get the data required for rendering the Stripe Elements, the application needs to call the `InitializePreOrderPayment` Glue API endpoint.
+
+#### Example of rendering Stripe Elements on the summary page
+
+```JS
+
+async initializePreOrderPayment() {
+    const requestData = {
+      data: {
+        type: 'payments',
+        attributes: {
+          quote: QUOTE_DATA,
+          payment: {
+            amount: GRAND_TOTAL, // You will get it through the `/checkout-data?include=carts` endpoint
+            paymentMethodName: 'stripe', // taken from /checkout-data?include=payment-methods
+            paymentProviderName: 'stripe',  // taken from /checkout-data?include=payment-methods
+          },
+          preOrderPaymentData: {
+            "transactionId": this.transactionId, // This is empty in the first request but has to be used in further requests
+          },
+        },
+      },
+    };
+
+    const responseData = await this.fetchHandler(`GLUE_APPLICATION_URL/payments`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ACCESS_TOKEN`,
+      },
+      body: JSON.stringify(requestData),
+    });
+
+    const paymentProviderData =
+      responseData.data.attributes.preOrderPaymentData;
+
+    this.transactionId = paymentProviderData.transactionId;
+    this.accountId = paymentProviderData.accountId; // only be used on the Direct business model. When using a Marketplace business model this will not be present.
+
+    await this.setupStripe();
+  }
+
+```
+
+To identify the customer, you can use the `Authorization` and `X-Anonymous-Customer-Unique-Id` headers.
```

Comment:
Why do I need to identify the customers? How is it related to the code above?

Reply:
This is usual in Glue checkouts. Either you have a logged-in customer or a guest customer.